cairo/ethereum/cancun/fork.cairo

```diff
@@ -311,8 +313,20 @@ func process_transaction{
         assert is_valid.value = TRUE;
     }
 
-    // Get sender account
+    // Set original_storage_tries
     let state = env.value.state;
+    let storage_tries = state.value._storage_tries;
+    let original_storage_tries = copy_TrieTupleAddressBytes32U256{trie=storage_tries}();
```

Comment:
We need a new logic to properly squash / merge this new "read only" trie -> @ClementWalter calls it "rebase" -> when you are at the end of the tx, you need to rebase the final storage trie on the original trie, such that when you squash, all the read operations of original state trie are coherent